subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/OutputFilePropertyType.java

```diff
@@ -21,25 +21,30 @@
 
 @UsedByScanPlugin("test-distribution")
 public enum  OutputFilePropertyType {
```

Comment:
We could remove the whole enum, though it is currently used by the test distribution plugin since they also visit the task properties.

Reply:
You'd still need to make a difference between `FILE` and `DIRECTORY`, no?

Reply:
Yes, though we have `TreeType` for this already.

Reply:
Or maybe we still need the enum, since the unpacking code depends on type of the property (plural or singular).